### PR TITLE
MachinePool default Spec.Replicas

### PR DESCRIPTION
**Pull request by nprokopic**
Setting default value for `Spec.Replicas` to 1, as specified in Cluster API.

Default value is set on:
- `MachinePool` create, when `Spec.Replicas` is not specified,
- `MachinePool` update, when `Spec.Replicas` field has been set to `nil` (so we set it back to the default value).

### CODE REVIEW

File: pkg/machinepool/mutate_machinepool_create.go [hunk 1]
```diff
@@ -39,11 +39,16 @@ func (m *CreateMutator) Mutate(ctx context.Context, request *v1beta1.AdmissionRe
 		return result, nil
 	}
 
-	machinePoolCR := &capiv1alpha3.Cluster{}
+	machinePoolCR := &capiexp.MachinePool{}
```
Comment (whites11) [inline pkg/machinepool/mutate_machinepool_create.go:42]: Was this a bug on master?

**Review by whites11 — APPROVED** [review]
LGTM when tested

File: pkg/machinepool/mutate_machinepool_create.go (hunk 1 above)
Comment (nprokopic) [reply pkg/machinepool/mutate_machinepool_create.go:42]: yup :)

File: pkg/machinepool/defaults.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	}
+
+	return patches
+}
+
+// setDefaultReplicaValue checks if Spec.Replicas has been set, and if it is
+// not, it sets its value to 1.
+func (m *CreateMutator) setDefaultReplicaValue(ctx context.Context, machinePool *capiexp.MachinePool) *mutator.PatchOperation {
+	if machinePool.Spec.Replicas == nil {
+		const defaultReplicas = "1"
```
Comment (fiunchinho) [inline pkg/machinepool/defaults.go:29]: We normally define the constants in the top of the file, but I'm not blocking. Your call.

**Review by fiunchinho — APPROVED** [review]

**Review by fiunchinho — CHANGES_REQUESTED** [review]
Update the changelog please

File: pkg/machinepool/defaults.go (hunk 2 above)
Comment (nprokopic) [reply pkg/machinepool/defaults.go:29]: I basically needed a local var, which is not used anywhere else, but then changed it to `const` without too much thinking, as it does not change nor it should change.

**Comment by nprokopic** [comment]
> Update the changelog please

Updated.

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**